packages/core/src/lib/filter/index.ts

```diff
@@ -357,6 +359,9 @@ class Filter extends BaseProtocol implements IReceiver {
     return subscription;
   }
 
+  //TODO: Remove when FilterCore and FilterSDK are introduced
```

Comment:
Maybe `FitlerRFC` and `Filter` or `FilterCore` and `Filter`? I think one can be left clean or renamed entirely as it is something else than `Filter` in the meaning of protocol.

Reply:
I'd like to avoid this kind of names in general as it is kind of duplication when we have `FilterX` and `FilterY` or `SendResult` and `CoreSendResult` which just brings difficulty to understand what to use under which conditions. 

Ideally we don't want anyone use core protocols except when people know what they are doing. In that case for end user I'd like to keep out API the same: `LightPush`/`Filter` or create new entities like `Sender`/`Receiver`. That way proposition of implementation is clear. 

wdyt?

Reply:
> Maybe `FitlerRFC` and `Filter` or `FilterCore` and `Filter`? 
I personally incline towards a more verbose naming convention, but I'm happy to change this to naming like `Filter` and `FilterCore` if you prefer

> I think one can be left clean or renamed entirely as it is something else than `Filter` in the meaning of protocol.
Can you elaborate?

Reply:
> I personally incline towards a more verbose naming convention

I think move simpler names make it easier to comprehend and comply with RFC and other implementations.
For example - `FilterRFC` is visible that follows `RFC` and `Filter` being exposed at the root level of `node` makes it go-to choice with it's API being documented.  

> I think one can be left clean or renamed entirely as it is something else than Filter in the meaning of protocol.

here I mean that if we have `FilterRFC` being implementation of `RFC` - then `Filter` with it's proxy capabilities might as well be named something else, as it is not `Filter` in sense of Waku terminology but convenient `API` we provide

Reply:
RFC stands for Request For Comments - I'm not sure if the class called `FilterRFC` gives the right messaging.
We could, however, keep `FilterCore` as `Filter` and `FilterSDK` could remain as is.

Reply:
Since this was merged, @weboko do we need to followup here? Can be added as a separate task that can be tackled in #1886

Reply:
let's not add any subtasks for now - as we don't know what we are optimizing for